tests/net/lib/tls_credentials/src/tls_config/user-tls-conf-csr.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* MbedTLS custom configuration file for use with prj-trusted-tfm-csr.conf */
+
+#define MBEDTLS_X509_CSR_PARSE_C
```

Comment:
Question: can't we just enable `CONFIG_MBEDTLS_X509_CSR_PARSE_C` in the build instead of adding a new header file with basically only a single line?

Reply:
I apologize, I don't know how this version of the header file even got committed. 

The current version is 

```c
/* Use PSA */
#define MBEDTLS_USE_PSA_CRYPTO

/* X509 read/write and dependencies */
#define MBEDTLS_X509_USE_C
#define MBEDTLS_X509_CSR_WRITE_C
#define MBEDTLS_X509_CREATE_C
#define MBEDTLS_X509_CSR_PARSE_C
#define MBEDTLS_ASN1_WRITE_C
#define MBEDTLS_ASN1_PARSE_C
#define MBEDTLS_OID_C

/* Private Key support */
#define MBEDTLS_PK_PARSE_C
#define MBEDTLS_PK_WRITE_C
#define MBEDTLS_PK_C
```

and AFAICT all of these are required.

But I am not the best at dealing with these config headers. Are you able to spot a KConfig-based way to get all of these?